### PR TITLE
split AggInit into multiple op-specific specials

### PR DESCRIPTION
**Pull request by mars1026**

### CODE REVIEW

File: tile/codegen/aggrinit.cc [hunk 1]
```diff
@@ -1,5 +1,7 @@
 // Copyright 2018, Intel Corporation
 
+#include <memory>
+#include <set>
 #include <vector>
```
Comment (LLITCHEV) [inline tile/codegen/aggrinit.cc]: Did we need to include these?

File: tile/codegen/aggrinit.cc [hunk 2]
```diff
@@ -78,7 +80,7 @@ void AggregationBlockOutputInitializationPass::Apply(CompilerState* state) const
   for (const auto& toInit : this->state.blocksWithInits) {
     assert(toInit.blockToAddTo != nullptr);
     assert(toInit.refToInitialize != nullptr);
-    assert(toInit.initType == AggregationInitType::ADD || toInit.initType == AggregationInitType::MULL ||
+    assert(toInit.initType == AggregationInitType::ADD || toInit.initType == AggregationInitType::MUL ||
```
Comment (LLITCHEV) [inline tile/codegen/aggrinit.cc]: I think this change needs to change the definition of the AggregationInitType::MUL and I seem to miss that change.

File: tile/codegen/aggrinit.cc (hunk 1 above)
Comment (mars1026) [reply tile/codegen/aggrinit.cc]: cpplint complained about their absence, due to use of set<> and make_shared<>.

File: tile/codegen/aggrinit.cc (hunk 2 above)
Comment (LLITCHEV) [reply tile/codegen/aggrinit.cc]: Oh... that was the in assert and it is a typo. NM

**Pull request merged**